src/nvim/mark.c

```diff
@@ -979,6 +976,7 @@ void ex_delmarks(exarg_T *eap)
         for (i = from; i <= to; ++i) {
           if (lower) {
             curbuf->b_namedm[i - 'a'].mark.lnum = 0;
+            curbuf->b_namedm[i - 'a'].timestamp = os_time();
```

Comment:
I would get the value of `os_time()` only once, just before the `for (p = (char_u *)eap->arg; *p != NUL; p++)` loop